src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallerFactory.cs

```diff
@@ -20,7 +22,32 @@ internal class WorkloadInstallerFactory
             string dotnetDir = null, 
             PackageSourceLocation packageSourceLocation = null)
         {
+            var installType = GetWorkloadInstallType(sdkFeatureBand);
+
+            if (installType == InstallType.Msi && OperatingSystem.IsWindows())
```

Comment:
Should we error if the install type is MSI and the OS isn't windows? It seems like we've gotten into a bad state if that's the case.

Reply:
Right now it should just fall through to the file based installer. I did have code for it and then removed it, but we could throw something like an InvalidOperationException. This would be a weird state where someone had to create the sentinel file themselves on a non-Windows OS.

I can revive the code

Reply:
Personally I think it would be better to explicitly fail if we're in that state, but I don't have strong feelings.

Reply:
I don't either :) But I'll add the message back for now.